code/datums/components/object_possession.dm

```diff
@@ -0,0 +1,115 @@
+/// Component that allows admins to control any object as if it were a mob.
+/datum/component/object_possession
+	/// Stores a reference to the mob that is currently possessing the object.
+	var/datum/weakref/poltergeist = null
+	/// Ref to the screen object that is currently being displayed.
+	var/datum/weakref/screen_alert_ref = null
+	/**
+	  * back up of the real name during admin possession
+	  *
+	  * If an admin possesses an object it's real name is set to the admin name and this
+	  * stores whatever the real name was previously. When possession ends, the real name
+	  * is reset to this value
+	  */
+	var/stashed_name = null
+	/// List of signals we register to in order to know when to end possession.
+	var/static/list/signals_to_delete_on = list(
+		COMSIG_END_OBJECT_POSSESSION_VIA_COMPONENT_CHAIN,
+		COMSIG_END_OBJECT_POSSESSION_VIA_SCREEN_ALERT,
+		COMSIG_END_OBJECT_POSSESSION_VIA_VERB,
```

Comment:
Yeah these are... not good signals, I can't imagine any world where these signify generic events that I can hook into

Reply:
That makes this a bad component, which is the problem

Maybe this component should be on the mob controlling it, with a reference to what is being possessed?